### PR TITLE
Convert to Typescript

### PR DESCRIPTION
**Pull request by rhuanjl**
This is WIP but thought I'd open a PR for visibility.

EDIT: change of plan, forget jsdoc annotations, how about full typescript with strict mode enabled - only 3 errors all for unused parameters in incomplete functions.

### CODE REVIEW

File: oozaru.ts [hunk 1]
```diff
@@ -44,9 +44,9 @@ async function main()
 	Pegasus.initializeGlobals();
 	mainCanvas.onclick = () => {
 		mainCanvas.onclick = null;
-		let divElement = document.getElementById('prompt');
+		const divElement = <HTMLElement>document.getElementById('prompt');
```
Comment (fatcerberus) [inline oozaru.ts:47]: Do you actually need this cast?

File: oozaru/event-loop.ts [hunk 2]
```diff
@@ -30,60 +30,90 @@
...
-import from from './from.js';
 import * as galileo from './galileo.js';
 
 let
 	nextJobID = 1;
 
+export enum JobType {
+	update,
+	render,
+	immediate
```
Comment (fatcerberus) [inline oozaru/event-loop.ts:41]: Nit: Use PascalCase for enum members

**Review by fatcerberus — COMMENTED** [review]
I’ll give a more thorough review later perhaps.

File: oozaru.ts (hunk 1 above)
Comment (rhuanjl) [reply oozaru.ts:47]: Not 100% if a cast is the right answer - without it TSC errors that divElement may be null when you first try and use it.

**Comment by fatcerberus** [comment]
TypeScript definitely seems like a good fit for a project of this scope.  I’m a bit bothered by the seeming need for type casts in places where they don’t seem necessary.

It might just be that TSC knows an operation can fail and return `null` as a result, so it expects you to handle that case.  I don’t actually have much experience with TS but I know the type checker tries to be clever in cases where values can be more than one type.

**Comment by fatcerberus** [comment]
@rhuanjl I went on a refactoring binge last night and seem to have created a massive merge conflict. ☹️   If you can get this up-to-date again, I'll merge it ASAP and fix any style issues myself.

**Pull request closed without merging**